src/code/views/node-view.coffee

```diff
@@ -12,6 +12,9 @@ NodeTitle = React.createFactory React.createClass
   displayName: "NodeTitle"
   mixins: [require '../mixins/node-title']
 
+  componentWillMount: ->
+    @canDeleteWhenEmpty = @props.node.addedThisSession and not @props.node.titleUpdated
```

Comment:
Yes, `titleUpdated` isn't set until a change occurs and it is noted on line 45 - before that is is `undefined`.  So `canDeleteWhenEmpty` is true when a node is added during the session and the user first edits the title (which happens automatically).  The blur event on this component sets `titleUpdated` to true so on subsequence edits of the title a backspace or delete doesn't delete it.